spec/acceptance/reporting_spec.rb

```diff
@@ -55,54 +55,124 @@
     master.idempotent_apply(setup_manifest)
   end
 
-  context 'with report status: unchanged' do
-    context 'and noop_pending: false' do
-      let(:sitepp_content) do
-        # Puppet should report that nothing happens
-        ''
+  context 'with default incident_creation_report_attributes' do
+    context 'with report status: unchanged' do
+      context 'and no pending changes' do
+        let(:sitepp_content) do
+          # Puppet should report that nothing happens
+          ''
+        end
+
+        include_examples 'no incident'
       end
 
-      include_context 'reporting test setup'
+      context 'and pending changes' do
+        let(:sitepp_content) do
+          to_manifest(declare('notify', 'foo', 'noop' => true))
+        end
+
+        include_context 'incident creation test setup'
 
-      it 'does nothing' do
-        num_incidents_before_puppet_run = IncidentHelpers.get_incidents('').length
-        trigger_puppet_run(master, acceptable_exit_codes: [0])
-        num_incidents_after_puppet_run = IncidentHelpers.get_incidents('').length
-        expect(num_incidents_after_puppet_run).to eql(num_incidents_before_puppet_run)
+        include_examples 'no incident'
       end
     end
 
-    context 'and noop_pending: true' do
+    context 'with report status: intentional changes' do
+      # Default parameters report on corrective changes, not intentional changes.
       let(:sitepp_content) do
-        to_manifest(declare('notify', 'foo', 'noop' => true))
+        to_manifest(declare('notify', 'foo'))
       end
 
       include_context 'incident creation test setup'
-      include_examples 'incident creation test', 'noop_pending'
+      include_examples 'no incident'
     end
-  end
 
-  context 'with report status: changed' do
-    let(:sitepp_content) do
-      to_manifest(declare('notify', 'foo'))
+    context 'with report status: corrective changes' do
+      include_context 'incident creation test setup'
+      include_context 'corrective change'
+      include_examples 'incident creation test', 'changed'
     end
 
-    include_context 'incident creation test setup'
-    include_examples 'incident creation test', 'changed'
+    context 'with report status: failed changes' do
+      let(:sitepp_content) do
+        to_manifest(declare('exec', 'foo', 'command' => '/bin/foo_command'))
+      end
+
+      include_context 'incident creation test setup'
+      include_examples 'incident creation test', 'failed'
+    end
   end
 
-  context 'with report status: failed' do
-    let(:sitepp_content) do
-      to_manifest(declare('exec', 'foo', 'command' => '/bin/foo_command'))
+  context 'with user-specified incident_creation_report_statuses' do
+    context 'with pending changes reporting enabled' do
+      let(:params) do
+        super().merge('incident_creation_report_attributes' => ['corrective_changes', 'pending_changes'])
+      end
+
+      context 'with a pending corrective change' do
```

Comment:
The fact that this test passes feels like a bug in Puppet. I'll see what Platform says. Depending on what they say, we may need to remove this test b/c this should be impossible to detect with the current reporting API.

Reply:
Meh let's keep it. Can remove later if we need to. Not worth blocking the PR on it.